src/Command/CommandBase.php

```diff
@@ -472,19 +487,24 @@ protected function promptChooseLogs(
    * The list is generated from an array of objects. The objects much have at least one unique property and one
    * property that can be used as a human readable label.
    *
-   * @param object[]|ArrayObject $items An array of objects.
+   * @param object[]|array[] $items An array of objects or arrays.
    * @param string $unique_property The property of the $item that will be used to identify the object.
    * @param string $label_property
    * @param string $question_text
    *
    * @param bool $multiselect
    *
-   * @return null|object|array
+   * @return null|array|object
    */
-  public function promptChooseFromObjects($items, $unique_property, $label_property, $question_text, $multiselect = FALSE) {
+  public function promptChooseFromObjectsOrArrays($items, string $unique_property, string $label_property, string $question_text, $multiselect = FALSE) {
     $list = [];
     foreach ($items as $item) {
-      $list[$item->$unique_property] = trim($item->$label_property);
+      if (is_array($item)) {
+        $list[$item[$unique_property]] = trim($item[$label_property]);
```

Comment:
Is there a universal way to access a property that works for both arrays and objects?